testplan/runnable/interactive/http.py

```diff
@@ -429,6 +444,45 @@ def put(self, test_uid, suite_uid, testcase_uid, param_uid):
                 param_group[param_uid] = new_testcase
                 return param_group[param_uid].serialize()
 
+    @api.route("/report/export")
+    class ExportReport(flask_restplus.Resource):
+        """
+        Interactive export endpoint. There is an API for exporting root
+        report object.
+        """
+
+        def get(self):
+            available_exporters = []
+            with ihandler.report_mutex:
+                for exporter in ihandler.parent.exporters:
```

Comment:
I would push this logic to the interactive handler. The http handler should not know about that handler's internal, like ti has a parent, not talking about assumptions of aprent internal. 

The same apply to the history, and the logic calling the exporters itself.